### PR TITLE
makes rattlesnakes tamable

### PR DESCRIPTION
**Pull request by colossusqw**
[LABEL][Feature][AI]
## About the PR 
This PR makes the rattlesnakes tamable with food bits like scorpions.

## Why's this needed? 
Adds some fun critter interactions, for those willing to take the ristk. This was also requested multiple times in ghost chat and in the discord by other players so here it is.

```changelog
(u)Hans
(*)You can now tame rattlesnakes by giving them food, try not to get bit too many times trying.
```

### CODE REVIEW

File: code/obj/critter/misc.dm [hunk 1]
```diff
@@ -1554,6 +1555,40 @@
...
+			eat_twitch(src)
+		else
+			..()
+
+	attack_hand(mob/M)
+		if ((M.a_intent != INTENT_HARM) && (M in src.friends))
+			if(M.a_intent == INTENT_HELP && src.aggressive)
+				src.visible_message("<span class='notice'>[M] pats [src] on the head in a soothing way. It won't attack anyone now.</span>")
+				src.aggressive = FALSE
+				src.mobile = 1
```
Comment (frawhst) [inline code/obj/critter/misc.dm:1581]:
```suggestion
				src.mobile = TRUE
```
(at least, I assume this is a boolean)

File: code/obj/critter/misc.dm [hunk 2]
```diff
@@ -1554,6 +1555,40 @@
 				src.task = "chasing"
 				break
 
+	attackby(obj/item/W, mob/M)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
+			if(prob(25))
+				src.visible_message("<span class='notice'>[src] munches happily on the [W], and seems a little friendlier with [M]!</span>")
+				friends += M
```
Comment (frawhst) [inline code/obj/critter/misc.dm:1562]:
```suggestion
				src.friends += M
```
consistency

File: code/obj/critter/misc.dm [hunk 3]
```diff
@@ -1554,6 +1555,40 @@
...
+	attack_hand(mob/M)
+		if ((M.a_intent != INTENT_HARM) && (M in src.friends))
+			if(M.a_intent == INTENT_HELP && src.aggressive)
+				src.visible_message("<span class='notice'>[M] pats [src] on the head in a soothing way. It won't attack anyone now.</span>")
+				src.aggressive = FALSE
+				src.mobile = 1
+				icon_state = "rattlesnake"
+				src.task = "thinking"
+				return
+			else if((M.a_intent == INTENT_DISARM || M.a_intent == INTENT_GRAB) && !src.aggressive)
```
Comment (frawhst) [inline code/obj/critter/misc.dm:1585]: does this mean that punching a tamed rattlesnake won't make it mad, but disarming or grabbing it will

Comment (colossusqw) [reply code/obj/critter/misc.dm:1585]: it means you can make it aggressive towards other people like it is with scorpions

**Review by Tarmunora — APPROVED** [review]
the intent-based aggression toggle stuff is pretty clunkly, but otherwise looks alright

File: code/obj/critter/misc.dm (hunk 3 above)
Comment (frawhst) [reply code/obj/critter/misc.dm:1585]: Ah gotcha

**Review by frawhst — APPROVED** [review]
Intent thing is fine right now. Could be a context action or something in the future

**Pull request merged**